src/experimental/constructs/error-budget-alarm.ts

```diff
@@ -0,0 +1,207 @@
+}
+
+interface MonitorBurnRateForPeriodProps {
+  alarmName: string;
+  burnRateSpeed: BurnRate;
+  period: Duration;
+  errorBudget: number;
+  badEvents: IMetric;
+  validEvents: IMetric;
+}
+
+interface CompositeBurnRateAlarmProps {
+  sloName: string;
+  burnRateMonitoring: BurnRateMonitoring;
+  errorBudget: number;
+  badEvents: IMetric;
+  validEvents: IMetric;
+  snsTopic: ITopic;
+  suppressorAlarm?: CompositeBurnRateAlarm;
+}
+
+/**
+ * Use this construct to create alarms that will notify you when you are at risk of missing your SLO targets.
+ *
+ * These alarms follow the recommendations in the [SRE Workbook](https://sre.google/workbook/alerting-on-slos/) to
+ * implement Multiwindow, Multi-Burn-Rate Alerts.
+ *
+ * The CloudWatch implementation of this strategy creates nine separate CloudWatch alarms. The main alarms that users
+ * need to care about are the three composite alarms. These track different burn rates of error budget consumption:
+ * fast (e.g. complete outages), medium (e.g. moderate error percentage over hours, rather than minutes) and slow (e.g.
+ * a small percentage of errors sustained over a much longer period). Multiple burn rates are used to ensure that
+ * we are able to pick up all scenarios which could threaten the SLO, without interrupting developers with an
+ * unnecessary level of urgency.
+ *
+ * Each composite alarm creates two child alarms; these are used to support multiple windows (or periods in CloudWatch
+ * terminology). A composite alarm is configured to send notifications if both of its child alarms are in an alarm state.
+ * The child alarms themselves have no actions/notifications configured and users should not need to interact with them
+ * directly. Multiple windows are helpful for accuracy and reset time. The long window helps us to avoid sending an
+ * alert in scenarios where no intervention is required, for example a very brief spike in errors that is resolved
+ * automatically (e.g. when AWS replaces a 'bad' EC2 instance). The short window helps to ensure that the alert is reset
+ * (or moved back into OK state, in CloudWatch terminology) promptly once a problem has been resolved.
+ *
+ * The composite alarms have an awareness of priority, meaning that low priority alarm notifications are suppressed if
+ * a higher priority alarm is already firing. That is, if you receive an alert about fast error budget consumption, you
+ * should not receive an alert from the medium or slow versions of the alarm.
+ */
+export class ErrorBudgetAlarmExperimental {
+  // TODO: Should this extend Construct?
+
+  constructor(scope: GuStack, props: ErrorBudgetAlarmProps) {
+    const errorBudget = 1 - props.sloTarget;
+
+    const snsTopic: ITopic = Topic.fromTopicArn(
+      scope,
+      `SnsSloAlarmsFor${props.sloName}`,
+      `arn:aws:sns:${scope.region}:${scope.account}:${props.snsTopicNameForAlerts}`
+    );
+
+    const fastBurnRate: BurnRateMonitoring = {
+      speed: "Fast",
+      burnRate: 14.4,
```

Comment:
How have these burn rates been arrive at?

Reply:
Good question - I should have mentioned this in the PR description. Fast/medium are just based on [Google's recommendation](https://sre.google/workbook/alerting-on-slos/#recommended_parameters_for_an_slo_based_a), as that seems like a sensible place to start.

Unfortunately we cannot use their recommendation for slow burn rate due to a CloudWatch limitation (the maximum period that you can monitor with a CloudWatch alarm is 1 day), so I have used the other example that they give for a slow burn alarm instead for now.

Reply:
There is a strong scaling argument for keeping these burn-rates standardised as much as possible - as the number of alerts increases it becomes more and more confusing to manage different burn rates. 
One approach could be to have a standard construct which we treat as the recommended approach and a custom construct where burn rates could be manually set.
Another approach ,suggested by Google, is to class error budget alerts into different families or buckets and set standard burn specific to each family's needs. See [here](https://sre.google/workbook/alerting-on-slos/#alerting_at_scale) for details. Recommend we look at these approaches in future PR's though.